### PR TITLE
[kops-aws-platform] Disable kops-iam by default

### PR DESCRIPTION
**Pull request by Nuru**
## what
Disable kops iam integration by default

## why
Most users do not want kops iam integration, but prior to this change, it was forced on them.

### CODE REVIEW

File: aws/kops-aws-platform/iam-authenticator.tf [hunk 1]
```diff
@@ -1,6 +1,13 @@
+variable "kops_iam_enabled" {
+  type        = "string"
+  description = "Set to true to allow the module to create EFS resources"
```
Comment (aknysh) [inline aws/kops-aws-platform/iam-authenticator.tf]: update description (remove EFS)

File: aws/kops-aws-platform/iam-authenticator.tf [hunk 2]
```diff
@@ -115,19 +129,21 @@ data "aws_iam_policy_document" "admin" {
...
 }
 
 module "iam_authenticator_config" {
-  source                = "git::https://github.com/cloudposse/terraform-aws-kops-iam-authenticator-config.git?ref=tags/0.1.1"
+  enabled               = "${var.kops_iam_enabled}"
+  source                = "git::https://github.com/cloudposse/terraform-aws-kops-iam-authenticator-config.git?ref=tags/0.2.0"
   cluster_id            = "${var.cluster_id}"
   kube_config_path      = "${var.kube_config_path}"
-  admin_iam_role_arn    = "${aws_iam_role.admin.arn}"
+  admin_iam_role_arn    = "${element(concat(aws_iam_role.admin.*.arn, list("")), 0)}"
```
Comment (aknysh) [inline aws/kops-aws-platform/iam-authenticator.tf]:
```suggestion
  admin_iam_role_arn    = "${join("", aws_iam_role.admin.*.arn)}"
```

File: aws/kops-aws-platform/iam-authenticator.tf [hunk 3]
```diff
@@ -115,19 +129,21 @@ data "aws_iam_policy_document" "admin" {
...
   admin_k8s_username    = "${var.admin_k8s_username}"
   admin_k8s_groups      = "${var.admin_k8s_groups}"
-  readonly_iam_role_arn = "${aws_iam_role.readonly.arn}"
+  readonly_iam_role_arn = "${element(concat(aws_iam_role.readonly.*.arn, list("")), 0)}"
```
Comment (aknysh) [inline aws/kops-aws-platform/iam-authenticator.tf]:
```suggestion
  readonly_iam_role_arn = "${join("", aws_iam_role.readonly.*.arn)}"
```

File: aws/kops-aws-platform/iam-authenticator.tf (hunk 2 above)
Comment (Nuru) [reply aws/kops-aws-platform/iam-authenticator.tf]: This is the [canonical idiom from the Terraform docs](https://www.terraform.io/upgrade-guides/0-11.html#referencing-attributes-from-resources-with-count-0), and in public code I like to follow conventions when it is reasonable. I agree that `join` is shorter and cleaner, but is there a functional difference?

Comment (aknysh) [reply aws/kops-aws-platform/iam-authenticator.tf]: o functional difference (we just use it everywhere in our code)

**Pull request merged**